internal/deviations/deviations.go

```diff
@@ -72,20 +74,60 @@ import (
 	"github.com/openconfig/ondatra"
 )
 
+func matchRegex(regex string, deviceInfo string) bool {
+	match, err := regexp.MatchString(regex, deviceInfo)
+	if err != nil {
+		log.Errorf("Error with regex match %v", err)
+	} else if match {
+		return true
+	}
+	return false
+}
+
 // lookupDutDeviations returns the deviations for the specified dut.
 func lookupDUTDeviations(dut *ondatra.DUTDevice) *mpb.Metadata_Deviations {
+	metadataDeviations := &mpb.Metadata_Deviations{}
 	for _, platformExceptions := range metadata.Get().PlatformExceptions {
 		if dut.Device.Vendor().String() != platformExceptions.GetPlatform().Vendor.String() {
 			continue
 		}
-		for _, hardwareModel := range platformExceptions.GetPlatform().HardwareModel {
-			if dut.Device.Model() == hardwareModel {
+
+		hardwareModelRegex := platformExceptions.GetPlatform().GetHardwareModelRegex()
+		softwareVersionRegex := platformExceptions.GetPlatform().GetSoftwareVersionRegex()
+		hardwareModelCount := len(platformExceptions.GetPlatform().GetHardwareModel())
+
+		if hardwareModelRegex != "" {
+			// Return an error if both hardware_model_regex and repeated hardware_model fields are set.
+			if hardwareModelCount > 0 {
+				log.Fatalf("Cannot set repeated hardware model field: %v and \n Hardware model regex: %v", platformExceptions.GetPlatform().GetHardwareModel(), platformExceptions.GetPlatform().GetHardwareModelRegex())
```

Comment:
Suggested changes here:

- "Exitf" instead of "Fatalf" for a cleaner error message
- drop the word "repeated"
- "hardware model" -> "hardware_model"
- "Hardware model regex" -> "hardware_model_regex"

Reply:
done